### ISSUE TITLE
Wrong result of Domain.findAllByField(null)

### ISSUE THREAD

**Issue by demon101** [grails/grails-core#10730]
Thanks for reporting an issue for Grails framework, please review the task list below before submitting the 
issue. Your issue report will be closed if the issue is incomplete and the below tasks not completed.

NOTE: If you are unsure about something and the issue is more of a question a better place to ask questions is on Stack Overflow (http://stackoverflow.com/tags/grails) or Slack (http://slack-signup.grails.org). DO NOT use the issue tracker to ask questions.

### Task List

- [X] Steps to reproduce provided
- [ ] Example that reproduces the problem uploaded to Github
- [X] Full description of the issue provided (see below)

### Steps to Reproduce

For searching by null-value field dynamic finder in grails 3 works wrong 
**Domain.findAllByField(null)**  returns nothing. 
But 
**Domain.findAllByFieldIsNull()** works well
In my app field has enum type. 

### Expected Behaviour

Returns all records with an empty field from DB. In Grails 2.4.5 it works well

### Actual Behaviour

Returns empty list even if record presented in DB

### Environment Information

- **Grails Version:** 3.2.11
- **Gorm Version:** 6.0.12

**Comment by jeffbrown** [comment]
> Domain.findAllByField(null) returns nothing.

Does that mean it returns `null`, an empty `List`, or something else?

**Comment by demon101** [comment]
Hi @jeffbrown,
See attachment bootstrap. 
Bug happens only with sorting, returns empty list

[testinlist.zip](https://github.com/grails/grails-core/files/1156054/testinlist.zip)

**Comment by demon101** [comment]
@jeffbrown yes, workaround available, but the bug pretty lousy: GORM returns the wrong result. In my project this bug produced extraordinarily behaviour of usual code
